eng/pipelines/variables/common.yml

```diff
@@ -10,14 +10,21 @@ variables:
   value: false
 - name: publicGitRepoUri
   value: https://github.com/dotnet/dotnet-buildtools-prereqs-docker
-- name: manifestVariables
-  value: --var UniqueId=$(sourceBuildId)
+- ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/production') }}:
+  - name: manifestVariables
+    value: --var UniqueId=$(sourceBuildId) --var FloatingTagSuffix=latest
+- ${{ elseif eq(variables['Build.SourceBranch'], 'refs/heads/main') }}:
+  - name: manifestVariables
+    value: --var UniqueId=$(sourceBuildId) --var FloatingTagSuffix=staging
+- ${{ else }}:
+  - name: manifestVariables
+    value: --var UniqueId=$(sourceBuildId) --var FloatingTagSuffix=$(Build.SourceBranchName)
 - name: publicSourceBranch
-  value: main
+  value: $(Build.SourceBranchName)
```

Comment:
@mthalman @MichaelSimons I updated this branch name to be Build.SourceBranchName, so that for the production branch, this will be production (so we aren't lying)

Reply:
This will need to be changed in some way. The `publicSourceBranch` variable is used to determine which [image info file from the dotnet/version repo](https://github.com/dotnet/[versions](https://github.com/dotnet/versions/tree/main/build-info/docker)/tree/main/build-info/docker) is used. For example, this is now causing a failure in the PR build because `$(Build.SourceBranchName)` is equal to `refs/pull/667/merge`. So the pipeline logic is looking for an image info file named `image-info.dotnet-dotnet-buildtools-prereqs-docker-merge.json` which doesn't exist.

Since the intention is to now maintain multiple branches for this repo, this will need to be more dynamic. How about this:

```yaml
- ${{ if or(endsWith($(Build.SourceBranchName), "/production"), endsWith($(System.PullRequest.TargetBranch), "/production")) }}:
  - name: publicSourceBranch
    value: production
- ${{ else }}:
  - name: publicSourceBranch
    value: main
```

Before builds can proceed in the production branch, we'll need to have a corresponding file at https://github.com/dotnet/versions/blob/main/build-info/docker/image-info.dotnet-dotnet-buildtools-prereqs-docker-production.json.

Reply:
Should be $SourceBranch and not SourceBranchName the first if there.

```
- ${{ if or(eq($(Build.SourceBranch), "refs/heads/production"), eq($(System.PullRequest.TargetBranch), "refs/heads/production")) }}:
```

Reply:
Ah I didn;t understand what this was used for. Thanks for that. I will update

Reply:
Fixed.